src/entity/mesh.hpp

```diff
@@ -51,34 +53,36 @@ struct Mesh {
     // indices of vertices drawn (vertices used twice can be ignored)
     const std::vector<std::uint16_t> indices_;
     // position of vertices in mesh space
-    const std::vector<glm::vec3> indexed_vertices_;
+    const std::vector<glm::ivec3> indexed_vertices_;
     // color of vertex
-    const std::vector<glm::vec3> indexed_colors_;
+    const std::vector<uint16_t> indexed_color_ids_;
     // normal direction
-    const std::vector<glm::vec3> indexed_normals_;
+    const std::vector<glm::i8vec3> indexed_normals_;
+    // color map
+    const std::vector<uint32_t> color_map_;
 
 }; // struct Mesh
 
 /**
  * @brief Generates a mesh from the given 3D voxel structure
  *
- * @tparam T
+ * @tparam T is_base_of voxel_utility::VoxelLike, T
  * @param voxel_object
  */
 template <class T>
+requires std::is_base_of<voxel_utility::VoxelLike, T>::value
```

Comment:
First, define something like (notice I renamed the class `VoxelLike` to `VoxelBase`)
```cpp
template <class T>
concept VoxelLike = std::is_base_of<voxel_utility::VoxelBase, T>::value
```

Then change this to
```suggestion
template <VoxelLike T>
```